src/p_spec.c

```diff
@@ -2721,61 +2715,79 @@ void P_FreeScrollers (void)
 
 void R_InterpolateTextureOffsets (void)
 {
-  if (uncapped && leveltime > oldleveltime && !frozen_mode)
+  int i;
+
+  for (i = 0; i < numscrollers; i++)
   {
-    int i;
+    scroll_t *s = scrollers[i];
+    int dx, dy;
 
-    for (i = 0; i < numscrollers; i++)
+    if (s->accel)
     {
-      scroll_t *s = scrollers[i];
-      int dx, dy;
-
-      if (s->accel)
-      {
-        dx = s->vdx;
-        dy = s->vdy;
-      }
-      else
-      {
-        dx = s->dx;
-        dy = s->dy;
-
-        if (s->control != -1)
-        {   // compute scroll amounts based on a sector's height changes
-          fixed_t height = sectors[s->control].floorheight +
-            sectors[s->control].ceilingheight;
-          fixed_t delta = height - s->last_height;
-          dx = FixedMul(dx, delta);
-          dy = FixedMul(dy, delta);
-        }
+      dx = s->vdx;
+      dy = s->vdy;
+    }
+    else
+    {
+      dx = s->dx;
+      dy = s->dy;
+
+      if (s->control != -1)
+      {   // compute scroll amounts based on a sector's height changes
+        fixed_t height = sectors[s->control].floorheight +
+          sectors[s->control].ceilingheight;
+        fixed_t delta = height - s->last_height;
+        dx = FixedMul(dx, delta);
+        dy = FixedMul(dy, delta);
       }
+    }
 
-      if (!dx && !dy)
-        continue;
+    if (!dx && !dy)
+      continue;
 
-      switch(s->type)
-      {
-        side_t *side;
-        sector_t *sec;
+    switch(s->type)
+    {
+      side_t *side;
+      sector_t *sec;
 
-        case sc_side:
-          side = sides + s->affectee;
+      case sc_side:
+        side = sides + s->affectee;
+        if (uncapped && leveltime > oldleveltime && side->textureoffset != side->basetextureoffset)
           side->textureoffset = side->basetextureoffset + FixedMul(dx, fractionaltic);
+        else
```

Comment:
I think the problem here is that the old code does not support accumulation of delta values. That is, I think we should replace `side->textureoffset = side->basetextureoffset + FixedMul(dx, fractionaltic)` with `side->textureoffset += FixedMul(dx, fractionaltic)` here, but I have no chance to test this approach currently.

Reply:
With reduced game speed it doesn't work, don't know why. Maybe I messed something up. I'll try again later.